loader/standard_loader.py

```diff
@@ -13,58 +11,125 @@
+        self.bundles_failed_parsed: typing.List[ParsedBundle] = []
 
     @classmethod
     def _get_file_uuid(cls, file_guid: str):
         result = cls._uuid_regex.findall(file_guid.lower())
         if result is None:
-            raise ValueError(f'Misformatted file_guid: {file_guid} should contain a uuid.')
+            raise ParseError(f'Misformatted file_guid: {file_guid} should contain a uuid.')
         if len(result) != 1:
-            raise ValueError(f'Misformatted file_guid: {file_guid} contains multiple uuids. Only one was expected.')
+            raise ParseError(f'Misformatted file_guid: {file_guid} contains multiple uuids. Only one was expected.')
         return result[0]
 
-    @staticmethod
-    def _get_file_version(file_info: dict):
+    @classmethod
+    def _get_file_version(cls, file_info: dict):
         """Since date updated is optional, we default to date created when it's not updated"""
-        try:
-            return file_info['updated']
-        except KeyError:
-            return file_info['created']
+        def parse_version_key(file_info_, key):
+            """return None if version cannot be found"""
+            try:
+                match = cls._rfc3339_regex.fullmatch(file_info_[key])
+                if match is None:
+                    logger.warning(f'Failed to parse file version from date {key}: {file_info_[key]}')
+                    return None
+                return file_info_[key]
+            except KeyError:
+                return None
+        version = parse_version_key(file_info, 'updated')
+        if version is None:
+            version = parse_version_key(file_info, 'created')
+        if version is None:
+            raise ParseError('Either bundle had no updated / created time or it was not rfc3339 compliant')
+        return version
 
     @staticmethod
     def _get_cloud_urls(file_info: dict):
-        return {url_dict['url'] for url_dict in file_info['urls']}
+        if 'urls' not in file_info:
+            raise ParseError(f'URL field not present in file_info: \n{file_info}')
+        urls = file_info['urls']
+        if len(urls) < 1:
+            # FIXME: How many cloud URLs do we ACTUALLY need / expect?
+            raise ParseError(f'Expected at least one cloud url in file_info: \n{file_info}')
+        for url in urls:
+            if 'url' not in url:
+                raise ParseError(f"Expected 'url' as key for urls in file_info: \n{file_info}")
+        return [url_dict['url'] for url_dict in urls]
 
-    def _load_bundle(self, bundle_uuid, metadata_dict, data_objects):
+    @classmethod
+    def _parse_bundle(cls, bundle: dict) -> ParsedBundle:
+        try:
+            data_bundle = bundle['data_bundle']
+            bundle_uuid = data_bundle['id']
+            metadata_dict = data_bundle['user_metadata']
+            data_objects = bundle['data_objects']
+        except KeyError:
+            logger.exception('Failed to parse bundle')
```

Comment:
Is the exception message logged where the exception is caught and handled?
If so, as I expect it should be, then the message would not also need to be logged here as well.

Reply:
Good catch! It looks like there was duplicate logging.

Reply:
Actually, it's not quite duplicate since it's logging a different exception. Not sure whether or not to delete.

Reply:
On more experimentation, just copying the message from the first exception won't tell you much about what actually went wrong parsing. Instead I'm using the `raise ... from` pattern so it's clear what happened. LMK if you disagree.